public/js/components/RightSidebar.js

```diff
@@ -98,19 +103,29 @@ const RightSidebar = React.createClass({
     this.setupKeyboardShortcuts();
   },
 
-  shouldComponentUpdate(nextProps) {
-    return this.props.keyShortcuts !== nextProps.keyShortcuts;
-  },
-
   renderStepButtons() {
+    const className = this.props.pause ? "active" : "disabled";
     return [
-      debugBtn(this.resume, "resume", "active", "Click to resume (F8)"),
-      debugBtn(this.stepOver, "stepOver", "active", "Step Over (F10)"),
-      debugBtn(this.stepIn, "stepIn", "active", "Step In (F11)"),
```

Comment:
Hmm... if you click the pause/resume button while the debugger is not paused, it will tell the server to pause on the next expression.